pyqtorch/parametric.py

```diff
@@ -335,6 +398,27 @@ def extra_repr(self) -> str:
             f"control: {self.control}, target:{(self.target,)}, param:{self.param_name}"
         )
 
+    @cached_property
+    def eigenvals_generator(self) -> Tensor:
+        """Get eigenvalues of the underlying generator.
+
+        Arguments:
+            values: Parameter values.
+
+        Returns:
+            Eigenvalues of the generator operator.
+        """
+        return torch.cat(
+            (
+                torch.zeros(
+                    2 ** len(self.qubit_support) - 2,
+                    device=self.device,
+                    dtype=self.dtype,
+                ),
+                pauli_singleq_eigenvalues.flatten(),
```

Comment:
same here again, this will default to device 'cpu' and 'dtype' cdouble and error if the gate is on GPU and cfloat

Reply:
Fixed by also by `pauli_singleq_eigenvalues.flatten().to(...)` @dominikandreasseitz